OFI BTL compile broken with libfabric 1.5.0

Looks like the configure.m4 test needs to be expanded to also check for `FI_EINTR`:

```
btl_ofi_component.c: In function ‘mca_btl_ofi_component_progress’:
btl_ofi_component.c:563:63: error: ‘FI_EINTR’ undeclared (first use in this function)
         } else if (OPAL_UNLIKELY(ret != -FI_EAGAIN && ret != -FI_EINTR)) {
                                                               ^
../../../../opal/include/opal/prefetch.h:44:55: note: in definition of macro ‘OPAL_UNLIKELY’
 #define OPAL_UNLIKELY(expression) __builtin_expect(!!(expression), 0)
                                                       ^
btl_ofi_component.c:563:63: note: each undeclared identifier is reported only once for each function it appears in
         } else if (OPAL_UNLIKELY(ret != -FI_EAGAIN && ret != -FI_EINTR)) {
                                                               ^
../../../../opal/include/opal/prefetch.h:44:55: note: in definition of macro ‘OPAL_UNLIKELY’
 #define OPAL_UNLIKELY(expression) __builtin_expect(!!(expression), 0)
                                                       ^
```

--- a/opal/mca/btl/ofi/btl_ofi_component.c
+++ b/opal/mca/btl/ofi/btl_ofi_component.c
@@ -560,7 +560,15 @@ static int mca_btl_ofi_component_progress (void)
 
             MCA_BTL_OFI_ABORT();
 
-        } else if (OPAL_UNLIKELY(ret != -FI_EAGAIN && ret != -FI_EINTR)) {
+        }
+#ifdef FI_EINTR
+        /* sometimes, sockets provider complain about interupt. */
+        else if (OPAL_UNLIKELY(ret == -FI_EINTR)) {
+            continue;
+        }
+#endif
+        /* If the error is not FI_EAGAIN, report the error and abort. */
+        else if (OPAL_UNLIKELY(ret != -FI_EAGAIN)) {
             BTL_ERROR(("fi_cq_read returned error %d:%s", ret, fi_strerror(-ret)));
             MCA_BTL_OFI_ABORT();
         }

